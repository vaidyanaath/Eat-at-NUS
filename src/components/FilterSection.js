import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { StyledContainer } from './containers/StyledContainer';
import { InnerContainer } from './containers/InnerContainer';
import { RegularButton } from './buttons/RegularButton';
import { RegularText } from './texts/RegularText';
import { BigText } from './texts/BigText';

import SelectableChips from 'react-native-chip/SelectableChips';
import Slider from 'react-native-slider';
import { colors } from '../assets/colors';
import { SmallText } from './texts/SmallText';

export const FilterSection = () => {
  const [priceEnd, setPriceEnd] = React.useState(0);
  const [distance, setDistance] = React.useState(0);

  return (
    <StyledContainer style={styles.mainContainer}>
      <InnerContainer style={styles.headerContainer}>
        <BigText style={{ fontSize: 30 }}>Filter</BigText>
      </InnerContainer>
      <InnerContainer style={styles.cuisineContainer}>
        <RegularText style={styles.subHeading}>Cuisine</RegularText>
        <SelectableChips
          chipStyle={styles.chipContainer}
          valueStyle={styles.chipValue}
          chipStyleSelected={styles.chipSelectedContainer}
          valueStyleSelected={styles.chipSelectedValue}
          initialChips={[
            'Chinese',
            'Japanese',
            'Korean',
            'Thai',
            'Western',
            'Indian',
            'Malay',
            'Taiwanese',
          ]}
          onChangeChips={(chips) => console.log(chips)}
          alertRequired={false}
        />
      </InnerContainer>
      <InnerContainer style={styles.sliderContainer}>
        <InnerContainer style={styles.sliderHeader}>
          <RegularText style={styles.subHeadingRow}>Price</RegularText>
          <RegularText>&lt; ${priceEnd}</RegularText>
        </InnerContainer>
        <Slider
          onValueChange={(value) => setPriceEnd(value)}
          minimumValue={0}
          maximumValue={50}
          step={2}
          style={styles.slider}
        />
      </InnerContainer>

      <InnerContainer style={styles.sliderContainer}>
        <InnerContainer style={styles.sliderHeader}>
          <RegularText style={styles.subHeadingRow}>Distance</RegularText>
          <RegularText>&lt; {distance} m</RegularText>
        </InnerContainer>
        <Slider
          onValueChange={(value) => setDistance(value)}
          minimumValue={0}
          maximumValue={3000}
          step={100}
          style={styles.slider}
        />
      </InnerContainer>
    </StyledContainer>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    // alignItems: "center",
    // justifyContent: "center",
    // backgroundColor: '#ffc',
    borderRadius: 25,
    paddingTop: 10,
    paddingBottom: 20,
    paddingLeft: 30,
    paddingRight: 30,
  },
  cuisineContainer: {
    flex: 1,
    // alignItems: "flex-start",
    justifyContent: 'flex-start',
    // backgroundColor: "#ac3",
  },
  headerContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    maxHeight: 60,
    // backgroundColor: "#ffcdcc",
  },
  sliderContainer: {
    flex: 1,
    maxHeight: 80,
    // backgroundColor: '#cff',
  },
  sliderHeader: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  subHeading: {
    alignSelf: 'flex-start',
    marginVertical: 5,
  },
  subHeadingRow: {
    justifyContent: 'flex-start',
    alignSelf: 'center',
  },
  chipContainer: {
    maxHeight: 30,
    padding: 2,
    borderColor: colors.primary,
  },
  chipValue: {
    fontSize: 14,
    color: colors.primary,
  },
  chipSelectedContainer: {
    backgroundColor: colors.primary,
    borderColor: colors.primary,
  },
  chipSelectedValue: {
    color: colors.white,
    fontSize: 14,
  },
  slider: {
    flex: 1,
    width: "100%",
    // width: 200,
    // backgroundColor: "#23ada4",
  },

});
