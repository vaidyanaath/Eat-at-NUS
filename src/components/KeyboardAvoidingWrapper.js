import React from 'react';
import {
  KeyboardAvoidingView,
  ScrollView,
  TouchableWithoutFeedback,
  Keyboard,
  StyleSheet,
  View,
  Platform,
} from 'react-native';

export const KeyboardAvoidingWrapper = ({ children, style }) => {
  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior={Platform.OS === 'ios' ? 'padding' : null}
    >
      {/* <ScrollView contentContainerStyle={styles.scrollContainer}> */}
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <View style={style}>{children}</View>
        </TouchableWithoutFeedback>
      {/* </ScrollView> */}
    </KeyboardAvoidingView>
  );
};

// create styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  scrollContainer: {
    flex: 1,
  },
});
