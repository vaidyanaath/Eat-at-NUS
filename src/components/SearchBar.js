import React from 'react';
import { StyleSheet, View, TextInput, Image, TouchableOpacity } from 'react-native';

import { colors } from '../assets/colors';

export const SearchBar = () => {
    return(
      <View style={searchBarStyle.container}>
        <View style={searchBarStyle.searchContainer}>
          <Image source={require('../assets/images/search.png')} style={searchBarStyle.searchIcon} />
          <TextInput placeholder='Search' selectionColor={colors.secondary} style={searchBarStyle.searchTextInput}/>
        </View>
        <TouchableOpacity style={searchBarStyle.filterButton}>
        <Image source={require('../assets/images/filter.png')} style={searchBarStyle.filterIcon} />
        </TouchableOpacity>
  
      </View>
    );
  };
  
  const searchBarStyle = StyleSheet.create({
    container: {
      flex: 1, 
      flexDirection: 'row',
      minHeight: 50, 
      width: '100%',
      alignItems: 'center',
      justifyContent: 'space-between',
      // backgroundColor: '#e71837'
    },
    searchContainer : {
      flex: 1,
      flexDirection: 'row',
      minHeight: 20,
      marginRight: 10,
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 10,
      paddingHorizontal: 5,
      backgroundColor: colors.gray,
      paddingRight: 20,
    },
    searchIcon: {
      width: 40,
      height: 40,
      tintColor: colors.secondary,
      // backgroundColor: 'purple',
    },
    filterIcon: {
      width: 18,
      height: 18,
      tintColor: colors.gray,
    },
    filterButton: {
      backgroundColor: colors.primary,
      padding: 10,
      borderRadius: 10,
    },
    searchTextInput: {
      flex: 1,
      // backgroundColor: '#fed34a', // turquoise
    },
  
  });