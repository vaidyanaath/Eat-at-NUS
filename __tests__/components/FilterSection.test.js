import React from 'react';
import renderer from 'react-test-renderer';

import { FilterSection } from '../../src/components/FilterSection';

describe('<FilterSection />', () => {
    
    it('Renders correctly', () => {
        const tree = renderer.create(<FilterSection />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});