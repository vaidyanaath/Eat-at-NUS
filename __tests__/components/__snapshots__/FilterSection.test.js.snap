// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSection /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 30,
        "width": "100%",
      },
      Object {
        "borderRadius": 25,
        "flex": 1,
        "paddingBottom": 10,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxHeight": 50,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C365A",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 33,
          },
          Object {
            "fontSize": 30,
          },
        ]
      }
    >
      Filter
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "maxHeight": 160,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C365A",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 22,
          },
          Object {
            "alignSelf": "flex-start",
            "marginVertical": 5,
          },
        ]
      }
    >
      Cuisine
    </Text>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Chinese
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Japanese
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Korean
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Thai
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Western
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Indian
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Malay
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#2196F3",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "margin": 5,
                  "padding": 6,
                },
                Object {
                  "borderColor": "#ff7f50",
                  "maxHeight": 30,
                  "padding": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "color": "#2196F3",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#ff7f50",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Taiwanese
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "flex": 1,
          "maxHeight": 80,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 22,
            },
            Object {
              "alignSelf": "center",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        Price
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 22,
            },
          ]
        }
      >
        &lt; $
        0
      </Text>
    </View>
    <View
      animationType="timing"
      onLayout={[Function]}
      onValueChange={[Function]}
      step={2}
      style={
        Array [
          Object {
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "flex": 1,
            "width": "100%",
          },
        ]
      }
      thumbTouchSize={
        Object {
          "height": 40,
          "width": 40,
        }
      }
      value={0}
    >
      <View
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "#b3b3b3",
            },
            Object {
              "borderRadius": 2,
              "height": 4,
            },
            undefined,
          ]
        }
      />
      <View
        collapsable={false}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#3f3f3f",
            "borderRadius": 2,
            "height": 4,
            "opacity": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#343434",
            "borderRadius": 10,
            "height": 20,
            "opacity": 0,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
            "width": 20,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "flex": 1,
          "maxHeight": 80,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 22,
            },
            Object {
              "alignSelf": "center",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        Distance
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 22,
            },
          ]
        }
      >
        &lt; 
        0 m
         
      </Text>
    </View>
    <View
      animationType="timing"
      onLayout={[Function]}
      onValueChange={[Function]}
      step={100}
      style={
        Array [
          Object {
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "flex": 1,
            "width": "100%",
          },
        ]
      }
      thumbTouchSize={
        Object {
          "height": 40,
          "width": 40,
        }
      }
      value={0}
    >
      <View
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "#b3b3b3",
            },
            Object {
              "borderRadius": 2,
              "height": 4,
            },
            undefined,
          ]
        }
      />
      <View
        collapsable={false}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#3f3f3f",
            "borderRadius": 2,
            "height": 4,
            "opacity": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#343434",
            "borderRadius": 10,
            "height": 20,
            "opacity": 0,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
            "width": 20,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ff7f50",
        "borderRadius": 10,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C365A",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 22,
          },
          Object {
            "color": "#FFFFFF",
            "fontSize": 19,
          },
        ]
      }
    >
      Apply
    </Text>
  </View>
</View>
`;
